### PR TITLE
disable cleanup for inbound original dst cluster

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/51929

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### CODE REVIEW

File: pilot/pkg/networking/core/cluster_builder.go [hunk 1]
```diff
@@ -374,6 +377,11 @@ func (cb *ClusterBuilder) buildInboundCluster(clusterPort int, bind string,
 			string(instance.Service.Hostname), "", instance.Port.ServicePort, clusterPort, &instance.Service.Attributes)
 	}
 
+	if clusterType == cluster.Cluster_ORIGINAL_DST {
+		// Disable cleanup for inbound clusters - set to Max possible duration.
+		localCluster.cluster.CleanupInterval = durationpb.New(time.Duration(maxSecondsValue) * time.Second)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/cluster_builder.go:382]: is ·durationpb.New·  a special value, not quite underestand.

durationpb.New(1 year)?

**Comment by howardjohn** [comment]
cc @kyessenov I think you have configured this in the past?

File: pilot/pkg/networking/core/cluster_builder.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/networking/core/cluster_builder.go:382]: Any higher number is OK

**Comment by ramaraochavali** [comment]
> I think you have configured this in the past?

I think he has configured low idletimeout for pass through clusters but did not change the cleanup interval

**Comment by ramaraochavali** [comment]
@kyessenov Can you please take a look when you get chance? Ideally we do not need cleanup for inbound cluster and the regular default 1hr idle timeout should take care of cleaning up idle connections.  Let me know if you have any better suggestions?

**Comment by kyessenov** [comment]
@ramaraochavali I think it depends. From what I understand, this results in longer plaintext HTTP connections to the application ultimately. A poorly written application may leak memory for long connections, so resetting it often would address it. But a majority will probably benefit from less TCP handshakes?

**Comment by ramaraochavali** [comment]
> A poorly written application may leak memory for long connections, so resetting it often would address it.

Yes. But if it is continuously used, cleanup wont clean up. Also we have 1 hour idle timeout to take care of this.

>  But a majority will probably benefit from less TCP handshakes?
Yes

**Review by howardjohn — APPROVED** [review]

**Pull request merged**